product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/http/SOAPClient.java

```diff
@@ -109,6 +109,12 @@ public HttpResponse sendSimpleSOAPMessage(String url, final String payload, Stri
         HttpUriRequest request = new HttpPost(url);
         HTTPUtils.setHTTPHeader(SOAP_ACTION, action, request);
 
+        if(headers != null) {
```

Comment:
This method should not be called with a null headers map, either it should be empty, or a separate method which does not have the parameter 'headers' should be called.

Reply:
For the outside there is a separate method as you mentioned. They do not need to pass null. I am using same method to override the implementation.     
public HttpResponse sendSimpleSOAPMessage(String url, final String payload, String action) throws IOException 
        return sendSimpleSOAPMessage(url, payload, action, null);